### ISSUE TITLE
Controller not ready for SF3

### ISSUE THREAD

**Issue by clytemnestra** [sonata-project/SonataMediaBundle#1068]
#### Sonata packages

``` bash
$ composer show sonata-project/*
sonata-project/admin-bundle              3.3.2              The missing Symfony Admin Generator
sonata-project/block-bundle              3.0.1              Symfony SonataBlockBundle
sonata-project/cache                     1.0.7              Cache library
sonata-project/classification-bundle     3.0.0              Symfony SonataClassificationBundle
sonata-project/core-bundle               3.0.3              Symfony SonataCoreBundle
sonata-project/datagrid-bundle           2.2                Symfony SonataDatagridBundle
sonata-project/doctrine-extensions       1.0.2              Doctrine2 behavioral extensions
sonata-project/doctrine-orm-admin-bundle 3.0.4              Symfony Sonata / Integrate Doctrine ORM into the SonataAdminBundle
sonata-project/easy-extends-bundle       2.1.10             Symfony SonataEasyExtendsBundle
sonata-project/exporter                  1.5.0              Lightweight Exporter library
sonata-project/intl-bundle               2.2.4              Symfony SonataIntlBundle
sonata-project/media-bundle              dev-master 1eeae18 Symfony SonataMediaBundle
sonata-project/notification-bundle       3.0.0              Symfony SonataNotificationBundle
```
#### Symfony packages

``` bash
$ composer show symfony/*
symfony/monolog-bundle     2.11.1  Symfony MonologBundle
symfony/phpunit-bridge     v3.1.1  Symfony PHPUnit Bridge
symfony/polyfill-apcu      v1.2.0  Symfony polyfill backporting apcu_* functions to lower PHP versions
symfony/polyfill-intl-icu  v1.2.0  Symfony polyfill for intl's ICU-related data and classes
symfony/polyfill-mbstring  v1.2.0  Symfony polyfill for the Mbstring extension
symfony/polyfill-php56     v1.2.0  Symfony polyfill backporting some PHP 5.6+ features to lower PHP versions
symfony/polyfill-php70     v1.2.0  Symfony polyfill backporting some PHP 7.0+ features to lower PHP versions
symfony/polyfill-util      v1.2.0  Symfony utilities for portability of PHP codes
symfony/security-acl       v3.0.0  Symfony Security Component - ACL (Access Control List)
symfony/swiftmailer-bundle v2.3.11 Symfony SwiftmailerBundle
symfony/symfony            v3.1.1  The Symfony PHP framework
```
#### PHP version

``` bash
$ php -v
PHP 5.6.20 (cli) (built: Mar 31 2016 14:56:44) 
Copyright (c) 1997-2016 The PHP Group
Zend Engine v2.6.0, Copyright (c) 1998-2016 Zend Technologies
```
## Subject

The [MediaController](https://github.com/sonata-project/SonataMediaBundle/blob/3.x/Controller/MediaController.php) makes calls to `$this->get('request')` and `$this->getRequest()`, but in SF3 both of these calls are deprecated.

Are there any PRs to fix this? Because I couldn't find any, yet this seems a pretty big bug to me.And this is both on 3.x and master.
## Expected results

I think to solve it, add this to the controller

``` php
    /**
     * @return Request
     */
    public function getRequest()
    {
            return $this->get('request_stack')->getCurrentRequest();
    }
```

And replace `$this->get('request')` with `$this->get('request_stack')`.

The alternative is to add `(Request $request)` to the controller parameters.

**Comment by greg0ire** [comment]
This was solved in admin bundle. Would you provide a PR with the same solution?

**Comment by clytemnestra** [comment]
Should we replace all with ``$this->getRequest()`` with ``$this->get('request_stack')`` or should we also add the getRequest() function as well ?

**Comment by greg0ire** [comment]
The function is needed for BC I think

**Comment by clytemnestra** [comment]
But ``request_stack`` service is available in SF2.8, too. To what version you want the BC to go?

**Comment by Soullivaneuh** [comment]
@clytemnestra This is needed because the actual stable version support Symfony 2.3+.

**Comment by clytemnestra** [comment]
And that will be removed forever once you get everything stabled for SF3? At least that's what I understood, you don't want to maintain multiple branches anymore.

So we need to add that function in the controller.

**Comment by greg0ire** [comment]
That will be removed forever ~~once you get everything stabled for SF3~~ on the branch that corresponds to the next major (because we will drop old versions of sf on it).

**Comment by Soullivaneuh** [comment]
See also: https://github.com/sonata-project/dev-kit/issues/51

But the next major is not for right now, so SF 2.3 compatibility is needed.

**Comment by jordisala1991** [comment]
Fixed on latest release of MediaBundle 3.4.0
